### ISSUE TITLE
&shell suddenly set to sh instead of bash?

### ISSUE THREAD

**Issue by user202729** [lervag/vimtex#2845]
### Description

While using vimtex, I notice once that &shell is suddenly set to sh. My default setting is bash.

### Steps to reproduce

Currently not reproducible, but I suspect it's some sort of race condition in 
autoload/vimtex/jobs/vim.vim → vim_unix_run -- maybe if two files attempt to call the function independently, it will break...?

### VimtexInfo

Custom fork, actually. Might not be latest version, and I haven't tried reproducing it in latest version.

**Comment by user202729** [comment]
Anyway the reason why I suspect that is because the time I see that happen, I run `:9verb set shell` and see that that function is the culprit.

Is there any possibility the function can be called "asynchronously"? (i.e. is there some threading in vimscript?) Because if it's possible, it's certainly possible for two interleaving calls of the function to incorrectly restore the values:

```
thread A: backup saveshell = &shell (= bash)
thread A: set &shell = sh
thread B: backup saveshell = &shell (= sh)
etc.
```

then the original value (bash) is lost.

**Comment by lervag** [comment]
> Is there any possibility the function can be called "asynchronously"?

No, I don't think there is. Vimscript does not allow asynchronous stuff.

> … i.e. it if the `system()` call is _interrupted_ and the rest of the function is not executed? Is this possible?

Good question. It may be possible, but it would be _very_ unlikely!

**Comment by lervag** [comment]
I believe this issue was resolved by #2985.
